qiita_ware/private_plugin.py

```diff
@@ -335,6 +335,29 @@ def complete_job(job):
         job._set_status('success')
 
 
+def delete_analysis(job):
+    """Deletes a full analysis
+
+    Parameters
+    ----------
+    job : qiita_db.processing_job.ProcessingJob
+        The processing job performing the task
+    """
+    with qdb.sql_connection.TRN:
+        analysis_id = job.parameters.values['analysis_id']
+        analysis = qdb.analysis.Analysis(analysis_id)
+
+        artifacts = sorted(
+            analysis.artifacts, key=lambda a: a.id, reverse=True)
+
+        for artifact in artifacts:
+            qdb.artifact.Artifact.delete(artifact.id)
+
+        qdb.analysis.Analysis.delete(analysis.id)
+
+        job._set_status('success')
```

Comment:
Can you also delete the `'analysis_delete_%d'` key from redis once this job is completed? That will no longer be needed.